src/tests/tests_runtime.rs

```diff
@@ -299,3 +308,82 @@ fn test_features_disabled() {
         _ => panic!("Failed to run use_builtins.wasm"),
     }
 }
+
+#[test]
+#[serial]
+/// Non regression test on the AS class id values
+fn test_class_id() {
+    // setup basic AS runtime context
+    let interface: Box<dyn Interface> =
+        Box::new(TestInterface(Arc::new(Mutex::new(Ledger::new()))));
+    let bytecode = include_bytes!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/wasm/return_basic.wasm"
+    ));
+    let (module, engine) = ASModule::new(bytecode, 100_000, GasCosts::default()).unwrap();
+    let mut store = init_store(&engine).unwrap();
+    let mut context = ASContextModule::new(&*interface, module.binary_module, GasCosts::default());
+    let (instance, _) = context.create_vm_instance_and_init_env(&mut store).unwrap();
+
+    // setup test specific context
+    let (_, fenv) = context.resolver(&mut store);
+    let fenv_mut = fenv.into_mut(&mut store);
+    let memory = context.env.get_wasm_env().memory.as_ref().unwrap();
+    let memory_view = memory.view(&fenv_mut);
+
+    // get string and array offsets
+    let return_string = instance.exports.get_function("return_string").unwrap();
+    let return_array = instance.exports.get_function("return_array").unwrap();
+    let string_offset = return_string
```

Comment:
What is returned here is a pointer and not an offset...

Reply:
Yeah true I'll rename it

Reply:
Done

Reply:
I will exceptionally close this comment as we need to release the testnet @sydhds if you are not satisfied with the answers do not hesitate to open an issue